subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/cpp/plugins/CppTestConventionPlugin.java

```diff
@@ -0,0 +1,121 @@
+        project.getPluginManager().apply(CppBasePlugin.class);
+
+        final ConfigurationContainer configurations = project.getConfigurations();
+        final TaskContainer tasks = project.getTasks();
+
+        // TODO: What should this name be?
+        CppTestSuite component = objectFactory.newInstance(DefaultCppTestSuite.class, "test", objectFactory, fileOperations, configurations);
+        component.getBaseName().set("test");
+
+        // Register components created for the test component and test binaries
+        project.getComponents().add(component);
+        project.getComponents().add(component.getDevelopmentBinary());
+        // TODO: Need to register component as a project extension too
+
+        Action<Plugin<ProjectInternal>> projectConfiguration = new Action<Plugin<ProjectInternal>>() {
+            @Override
+            public void execute(Plugin<ProjectInternal> plugin) {
+                CppCompile compileMain = tasks.withType(CppCompile.class).getByName("compileDebugCpp");
+                CppCompile compileTest = tasks.withType(CppCompile.class).getByName("compileTestDebugCpp");
+
+                // TODO: This should probably be just the main component's public headers?
+                compileTest.includes(compileMain.getIncludes());
+
+                AbstractLinkTask linkTest = tasks.withType(AbstractLinkTask.class).getByName("linkTestDebug");
+                linkTest.source(compileMain.getObjectFileDir().getAsFileTree().matching(new PatternSet().include("**/*.obj", "**/*.o")));
+            }
+        };
+
+        project.getPlugins().withType(CppLibraryPlugin.class, projectConfiguration);
+        // TODO: We will get symbol conflicts with executables since they already have a main()
+        project.getPlugins().withType(CppExecutablePlugin.class, projectConfiguration);
+
+        // TODO: Replace with new native test task
+        final RunTestExecutable testTask = tasks.create("cpptest", RunTestExecutable.class, new Action<RunTestExecutable>() {
+            @Override
+            public void execute(RunTestExecutable testTask) {
+                testTask.setGroup(LifecycleBasePlugin.VERIFICATION_GROUP);
+                testTask.setDescription("Executes C++ tests.");
+
+                // TODO: It would be nice if the installation was a thing we could get path/dependencies from vs going to the task
+                final InstallExecutable installTask = (InstallExecutable) tasks.getByName("installTestDebug");
+                testTask.setExecutable(installTask.getRunScript());
+                testTask.dependsOn(installTask);
+
+                // TODO: This should be lazy to honor changes to the build directory
+                final DirectoryProperty buildDirectory = project.getLayout().getBuildDirectory();
+                testTask.setOutputDir(buildDirectory.dir("test-results/cpptest").get().getAsFile());
+            }
+        });
+
+        tasks.getByName("check", new Action<Task>() {
```

Comment:
not just `tasks.getByName("check").dependsOn(testTask)`?

Reply:
done